File: src/javaDocExamples/java/doc/java/examples/filters/MyApplicationCommandFilter.java
package doc.java.examples.filters;

import io.github.freya022.botcommands.api.commands.application.ApplicationCommandFilter;
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandInfo;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@BService
public class MyApplicationCommandFilter implements ApplicationCommandFilter {

    @Override
    public boolean getGlobal() {
        return true;
    }

    @Nullable
    @Override
    public String check(@NotNull GenericCommandInteractionEvent event, @NotNull ApplicationCommandInfo commandInfo) {
        if (event.getChannel().getIdLong() != 722891685755093076L) {
            event.reply("Can only run commands in <#722891685755093076>").setEphemeral(true).queue();
            return "Not the right channel";
        }
        return null;
    }
}

File: src/kotlinDocExamples/kotlin/doc/kotlin/examples/filters/MyApplicationCommandFilter.kt
package doc.kotlin.examples.filters

import dev.minn.jda.ktx.coroutines.await
import dev.minn.jda.ktx.messages.reply_
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandFilter
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandInfo
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent

@BService
class MyApplicationCommandFilter : ApplicationCommandFilter {

    override val global: Boolean get() = true

    override suspend fun checkSuspend(
        event: GenericCommandInteractionEvent,
        commandInfo: ApplicationCommandInfo
    ): String? {
        if (event.channel!!.idLong != 722891685755093076) {
            event.reply_("Can only run commands in <#722891685755093076>", ephemeral = true).await()
            return "Wrong channel"
        }
        return null
    }
}
